pkg/registry/remote.go

```diff
@@ -44,7 +46,15 @@ func readRemoteImage(n image.Name) (v1.Image, error) {
 		return nil, err
 	}
 
-	return remote.Image(ref, remote.WithAuth(auth))
+	img, err := remote.Image(ref, remote.WithAuth(auth))
+	if err != nil {
+		var derr error
+		img, derr = daemon.Image(ref)
+		if derr != nil {
+			return nil, fmt.Errorf("reading remote image %s failed: %v; attempting to read from daemon also failed: %v", n.String(), err, derr)
```

Comment:
I had to think carefully about the error message in this case. Initially, I ignored the daemon error and reported back just the repository error - the rationale being that since there is no hard dependency on the daemon, we should report errors from the daemon. However, this could mask a genuine problem with the daemon, so I decided to combine the two errors together, for example:
```
$ ./irel layout add layout glyn/helloworld-cnab:a028fb535114b81ae38a6711b5de896642a07dfd
2019/08/07 13:34:50 add failed: reading remote image docker.io/glyn/helloworld-cnab:a028fb535114b81ae38a6711b5de896642a07dfd failed: MANIFEST_UNKNOWN: manifest unknown; map[Tag:a028fb535114b81ae38a6711b5de896642a07dfd]; attempting to read from daemon also failed: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```